File: backup/src/jclang/CXFile.java


package jclang;

import com.sun.jna.PointerType;
import org.jetbrains.annotations.NotNull;
import jclang.structs.CXString;

import java.io.File;

public class CXFile extends PointerType {
    @NotNull
    public File toFile() {
        CXString.ByValue name = LibClang.I.getFileName(this);
        NativePool.I.record(name);
        return new File(LibClang.I.getCString(name));
    }
}

File: klang/libclang/src/main/kotlin/klang/jvm/binding/CXFile.kt
package klang.jvm.binding

import com.sun.jna.PointerType
import klang.jvm.Clang
import java.io.File

class CXFile : PointerType() {
    fun toFile(): File {
        return File(Clang.getFileName(this))
    }
}
